app/src/main/java/com/alexP/assignment1/ui/contactsActivity/ContactsActivity.kt

```diff
@@ -113,7 +110,7 @@ class ContactsActivity : BaseActivity<ActivityContactsBinding>() {
             }
 
             override fun onSwiped(viewHolder: RecyclerView.ViewHolder, direction: Int) {
-                deleteContact(viewHolder.itemView.tag as Contact)
+                deleteContact((viewHolder as ContactsAdapter.ContactsViewHolder).getContact())
```

Comment:
Since I already said to remove the getContact() function from the ViewHolder, there is another way to accomplish deleting a contact from the list. In this particular case, the order of contacts in the viewmodel field is equal to the order in which contacts are displayed in the recyclerview.
Your code will be look like
```kotlin
val contact = /*retrieve contact using the default stack of fields or properties of passed viewholder*/
deleteContact(contact) // or little bit transformed form that respects contacts nullability
```